### DISCUSSION TITLE
Setting email framework

### DISCUSSION THREAD

**Discussion by kriscr** [pimcore/pimcore#9690] [Q&A / Support]
Hi Team

How do you setup the email framework?

I followed the steps like described on https://pimcore.com/docs/pimcore/current/Development_Documentation/Development_Tools_and_Details/Email_Framework/index.html

```
framework:
    mailer:
        transports:
            main: smtp://mailhog:1025
            pimcore_newsletter: smtp://mailhog:1025
```

In my test, it always uses the sendmail_path from php.ini but not from the config.yml and this gives me an error 'Unsupported sendmail command flags ... '.

**Comment by dvesh3** [comment]
@kriscr It is based on Symfony Mailer. You need to download third party package to use it and please check the list of supported transports here: https://symfony.com/doc/current/mailer.html#using-a-3rd-party-transport

**Reply by kriscr** [reply]
Can you please send me your config?
If I type for example

```
framework:
    mailer:
        transports:
            main: gmail+smtp://USERNAME:PASSWORD@default
```
The Symfony Mailer uses the sendmail_path from php.ini instead of the params from config.yml file.

**Reply by nirbanc2020** [reply]
I used the same configuration in config.yml but I am getting error e.g
Handling "Symfony\Component\Mailer\Messenger\SendEmailMessage" failed: Failed to authenticate on SMTP server with username "pimcore.pgs@gmail.com" using the following authenticators: "LOGIN", "PLAIN", "XOAUTH2". Authenticator "LOGIN" returned "Symfony\Component\Mailer\Exception\TransportException: Expected response code "235" but got code "534", with message "534-5.7.14 534-5.7.14 Please log in via your web browser and then try again. 534-5.7.14 Learn more at 534 5.7.14 https://support.google.com/mail/answer/78754 az34sm7033072qkb.41 - gsmtp". in /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php:308 Stack trace: #0 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(182): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->assertResponseCode() #1 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/Auth/LoginAuthenticator.php(37): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->executeCommand() #2 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/EsmtpTransport.php(171): Symfony\Component\Mailer\Transport\Smtp\Auth\LoginAuthenticator->authenticate() #3 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/EsmtpTransport.php(136): Symfony\Component\Mailer\Transport\Smtp\EsmtpTransport->handleAuth() #4 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(253): Symfony\Component\Mailer\Transport\Smtp\EsmtpTransport->doHeloCommand() #5 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(194): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->start() #6 /var/www/html/siteone/vendor/symfony/mailer/Transport/AbstractTransport.php(71): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->doSend() #7 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(136): Symfony\Component\Mailer\Transport\AbstractTransport->send() #8 /var/www/html/siteone/vendor/symfony/mailer/Transport/Transports.php(51): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->send() #9 /var/www/html/siteone/vendor/symfony/mailer/Messenger/MessageHandler.php(31): Symfony\Component\Mailer\Transport\Transports->send() #10 /var/www/html/siteone/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php(96): Symfony\Component\Mailer\Messenger\MessageHandler->__invoke() #11 /var/www/html/siteone/vendor/symfony/messenger/Middleware/SendMessageMiddleware.php(74): Symfony\Component\Messenger\Middleware\HandleMessageMiddleware->handle() #12 /var/www/html/siteone/vendor/pimcore/portal-engine/src/Service/Messenger/Middleware/CollectGarbageMiddleware.php(29): Symfony\Component\Messenger\Middleware\SendMessageMiddleware->handle() #13 /var/www/html/siteone/vendor/symfony/messenger/Middleware/FailedMessageProcessingMiddleware.php(34): Pimcore\Bundle\PortalEngineBundle\Service\Messenger\Middleware\CollectGarbageMiddleware->handle() #14 /var/www/html/siteone/vendor/symfony/messenger/Middleware/DispatchAfterCurrentBusMiddleware.php(68): Symfony\Component\Messenger\Middleware\FailedMessageProcessingMiddleware->handle() #15 /var/www/html/siteone/vendor/symfony/messenger/Middleware/RejectRedeliveredMessageMiddleware.php(48): Symfony\Component\Messenger\Middleware\DispatchAfterCurrentBusMiddleware->handle() #16 /var/www/html/siteone/vendor/symfony/messenger/Middleware/AddBusNameStampMiddleware.php(37): Symfony\Component\Messenger\Middleware\RejectRedeliveredMessageMiddleware->handle() #17 /var/www/html/siteone/vendor/symfony/messenger/Middleware/TraceableMiddleware.php(43): Symfony\Component\Messenger\Middleware\AddBusNameStampMiddleware->handle() #18 /var/www/html/siteone/vendor/symfony/messenger/MessageBus.php(77): Symfony\Component\Messenger\Middleware\TraceableMiddleware->handle() #19 /var/www/html/siteone/vendor/symfony/messenger/TraceableMessageBus.php(41): Symfony\Component\Messenger\MessageBus->dispatch() #20 /var/www/html/siteone/vendor/symfony/mailer/Mailer.php(55): Symfony\Component\Messenger\TraceableMessageBus->dispatch() #21 /var/www/html/siteone/vendor/pimcore/pimcore/lib/Mail/Mailer.php(54): Symfony\Component\Mailer\Mailer->send() #22 /var/www/html/siteone/vendor/pimcore/pimcore/lib/Mail.php(571): Pimcore\Mail\Mailer->send() #23 /var/www/html/siteone/vendor/pimcore/pimcore/lib/Mail.php(505): Pimcore\Mail->sendWithoutRendering() #24 /var/www/html/siteone/vendor/pimcore/pimcore/bundles/AdminBundle/Controller/Admin/EmailController.php(433): Pimcore\Mail->send() #25 /var/www/html/siteone/vendor/symfony/http-kernel/HttpKernel.php(152): Pimcore\Bundle\AdminBundle\Controller\Admin\EmailController->sendTestEmailAction() #26 /var/www/html/siteone/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\Component\HttpKernel\HttpKernel->handleRaw() #27 /var/www/html/siteone/vendor/symfony/http-kernel/Kernel.php(202): Symfony\Component\HttpKernel\HttpKernel->handle() #28 /var/www/html/siteone/public/index.php(36): Symfony\Component\HttpKernel\Kernel->handle() #29 {main}". Authenticator "PLAIN" returned "Symfony\Component\Mailer\Exception\TransportException: Expected response code "235" but got code "534", with message "534-5.7.14 534-5.7.14 Please log in via your web browser and then try again. 534-5.7.14 Learn more at 534 5.7.14 https://support.google.com/mail/answer/78754 az34sm7033072qkb.41 - gsmtp". in /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php:308 Stack trace: #0 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(182): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->assertResponseCode() #1 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/Auth/PlainAuthenticator.php(35): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->executeCommand() #2 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/EsmtpTransport.php(171): Symfony\Component\Mailer\Transport\Smtp\Auth\PlainAuthenticator->authenticate() #3 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/EsmtpTransport.php(136): Symfony\Component\Mailer\Transport\Smtp\EsmtpTransport->handleAuth() #4 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(253): Symfony\Component\Mailer\Transport\Smtp\EsmtpTransport->doHeloCommand() #5 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(194): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->start() #6 /var/www/html/siteone/vendor/symfony/mailer/Transport/AbstractTransport.php(71): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->doSend() #7 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(136): Symfony\Component\Mailer\Transport\AbstractTransport->send() #8 /var/www/html/siteone/vendor/symfony/mailer/Transport/Transports.php(51): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->send() #9 /var/www/html/siteone/vendor/symfony/mailer/Messenger/MessageHandler.php(31): Symfony\Component\Mailer\Transport\Transports->send() #10 /var/www/html/siteone/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php(96): Symfony\Component\Mailer\Messenger\MessageHandler->__invoke() #11 /var/www/html/siteone/vendor/symfony/messenger/Middleware/SendMessageMiddleware.php(74): Symfony\Component\Messenger\Middleware\HandleMessageMiddleware->handle() #12 /var/www/html/siteone/vendor/pimcore/portal-engine/src/Service/Messenger/Middleware/CollectGarbageMiddleware.php(29): Symfony\Component\Messenger\Middleware\SendMessageMiddleware->handle() #13 /var/www/html/siteone/vendor/symfony/messenger/Middleware/FailedMessageProcessingMiddleware.php(34): Pimcore\Bundle\PortalEngineBundle\Service\Messenger\Middleware\CollectGarbageMiddleware->handle() #14 /var/www/html/siteone/vendor/symfony/messenger/Middleware/DispatchAfterCurrentBusMiddleware.php(68): Symfony\Component\Messenger\Middleware\FailedMessageProcessingMiddleware->handle() #15 /var/www/html/siteone/vendor/symfony/messenger/Middleware/RejectRedeliveredMessageMiddleware.php(48): Symfony\Component\Messenger\Middleware\DispatchAfterCurrentBusMiddleware->handle() #16 /var/www/html/siteone/vendor/symfony/messenger/Middleware/AddBusNameStampMiddleware.php(37): Symfony\Component\Messenger\Middleware\RejectRedeliveredMessageMiddleware->handle() #17 /var/www/html/siteone/vendor/symfony/messenger/Middleware/TraceableMiddleware.php(43): Symfony\Component\Messenger\Middleware\AddBusNameStampMiddleware->handle() #18 /var/www/html/siteone/vendor/symfony/messenger/MessageBus.php(77): Symfony\Component\Messenger\Middleware\TraceableMiddleware->handle() #19 /var/www/html/siteone/vendor/symfony/messenger/TraceableMessageBus.php(41): Symfony\Component\Messenger\MessageBus->dispatch() #20 /var/www/html/siteone/vendor/symfony/mailer/Mailer.php(55): Symfony\Component\Messenger\TraceableMessageBus->dispatch() #21 /var/www/html/siteone/vendor/pimcore/pimcore/lib/Mail/Mailer.php(54): Symfony\Component\Mailer\Mailer->send() #22 /var/www/html/siteone/vendor/pimcore/pimcore/lib/Mail.php(571): Pimcore\Mail\Mailer->send() #23 /var/www/html/siteone/vendor/pimcore/pimcore/lib/Mail.php(505): Pimcore\Mail->sendWithoutRendering() #24 /var/www/html/siteone/vendor/pimcore/pimcore/bundles/AdminBundle/Controller/Admin/EmailController.php(433): Pimcore\Mail->send() #25 /var/www/html/siteone/vendor/symfony/http-kernel/HttpKernel.php(152): Pimcore\Bundle\AdminBundle\Controller\Admin\EmailController->sendTestEmailAction() #26 /var/www/html/siteone/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\Component\HttpKernel\HttpKernel->handleRaw() #27 /var/www/html/siteone/vendor/symfony/http-kernel/Kernel.php(202): Symfony\Component\HttpKernel\HttpKernel->handle() #28 /var/www/html/siteone/public/index.php(36): Symfony\Component\HttpKernel\Kernel->handle() #29 {main}". Authenticator "XOAUTH2" returned "Symfony\Component\Mailer\Exception\TransportException: Expected response code "235" but got code "334", with message "334 eyJzdGF0dXMiOiI0MDAiLCJzY2hlbWVzIjoiQmVhcmVyIiwic2NvcGUiOiJodHRwczovL21haWwuZ29vZ2xlLmNvbS8ifQ==". in /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php:308 Stack trace: #0 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(182): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->assertResponseCode() #1 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/Auth/XOAuth2Authenticator.php(37): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->executeCommand() #2 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/EsmtpTransport.php(171): Symfony\Component\Mailer\Transport\Smtp\Auth\XOAuth2Authenticator->authenticate() #3 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/EsmtpTransport.php(136): Symfony\Component\Mailer\Transport\Smtp\EsmtpTransport->handleAuth() #4 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(253): Symfony\Component\Mailer\Transport\Smtp\EsmtpTransport->doHeloCommand() #5 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(194): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->start() #6 /var/www/html/siteone/vendor/symfony/mailer/Transport/AbstractTransport.php(71): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->doSend() #7 /var/www/html/siteone/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(136): Symfony\Component\Mailer\Transport\AbstractTransport->send() #8 /var/www/html/siteone/vendor/symfony/mailer/Transport/Transports.php(51): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->send() #9 /var/www/html/siteone/vendor/symfony/mailer/Messenger/MessageHandler.php(31): Symfony\Component\Mailer\Transport\Transports->send() #10 /var/www/html/siteone/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php(96): Symfony\Component\Mailer\Messenger\MessageHandler->__invoke() #11 /var/www/html/siteone/vendor/symfony/messenger/Middleware/SendMessageMiddleware.php(74): Symfony\Component\Messenger\Middleware\HandleMessageMiddleware->handle() #12 /var/www/html/siteone/vendor/pimcore/portal-engine/src/Service/Messenger/Middleware/CollectGarbageMiddleware.php(29): Symfony\Component\Messenger\Middleware\SendMessageMiddleware->handle() #13 /var/www/html/siteone/vendor/symfony/messenger/Middleware/FailedMessageProcessingMiddleware.php(34): Pimcore\Bundle\PortalEngineBundle\Service\Messenger\Middleware\CollectGarbageMiddleware->handle() #14 /var/www/html/siteone/vendor/symfony/messenger/Middleware/DispatchAfterCurrentBusMiddleware.php(68): Symfony\Component\Messenger\Middleware\FailedMessageProcessingMiddleware->handle() #15 /var/www/html/siteone/vendor/symfony/messenger/Middleware/RejectRedeliveredMessageMiddleware.php(48): Symfony\Component\Messenger\Middleware\DispatchAfterCurrentBusMiddleware->handle() #16 /var/www/html/siteone/vendor/symfony/messenger/Middleware/AddBusNameStampMiddleware.php(37): Symfony\Component\Messenger\Middleware\RejectRedeliveredMessageMiddleware->handle() #17 /var/www/html/siteone/vendor/symfony/messenger/Middleware/TraceableMiddleware.php(43): Symfony\Component\Messenger\Middleware\AddBusNameStampMiddleware->handle() #18 /var/www/html/siteone/vendor/symfony/messenger/MessageBus.php(77): Symfony\Component\Messenger\Middleware\TraceableMiddleware->handle() #19 /var/www/html/siteone/vendor/symfony/messenger/TraceableMessageBus.php(41): Symfony\Component\Messenger\MessageBus->dispatch() #20 /var/www/html/siteone/vendor/symfony/mailer/Mailer.php(55): Symfony\Component\Messenger\TraceableMessageBus->dispatch() #21 /var/www/html/siteone/vendor/pimcore/pimcore/lib/Mail/Mailer.php(54): Symfony\Component\Mailer\Mailer->send() #22 /var/www/html/siteone/vendor/pimcore/pimcore/lib/Mail.php(571): Pimcore\Mail\Mailer->send() #23 /var/www/html/siteone/vendor/pimcore/pimcore/lib/Mail.php(505): Pimcore\Mail->sendWithoutRendering() #24 /var/www/html/siteone/vendor/pimcore/pimcore/bundles/AdminBundle/Controller/Admin/EmailController.php(433): Pimcore\Mail->send() #25 /var/www/html/siteone/vendor/symfony/http-kernel/HttpKernel.php(152): Pimcore\Bundle\AdminBundle\Controller\Admin\EmailController->sendTestEmailAction() #26 /var/www/html/siteone/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\Component\HttpKernel\HttpKernel->handleRaw() #27 /var/www/html/siteone/vendor/symfony/http-kernel/Kernel.php(202): Symfony\Component\HttpKernel\HttpKernel->handle() #28 /var/www/html/siteone/public/index.php(36): Symfony\Component\HttpKernel\Kernel->handle() #29 {main}".

Server threw exception - could not perform action. Please reload the admin interface and try again.

**Reply by dvesh3** [reply]
@nirbanc2020 Either your password is wrong or setting "Allow less secure apps" is disabled in your gmail account.

**Comment by kriscr** [comment]
How can I use the Built-in Transports?

When using mail($to,$subject,$message, $headers); it works, but I want to use the Pimcore mail...
I still get 'Unsupported sendmail command flags "/usr/local/lib/mhsendmail.'
Can anyone help me?

**config.yml**
framework:
    mailer:
        transports:
            sendmail: //default

**Test script**
        $mail = new \Pimcore\Mail();
        $mail->to('example@pimcore.org');
        $mail->text("This is just plain text");
        $mail->send();

**Reply by dvesh3** [reply]
just use built-in for the main transport in config.yaml:
```
framework:
    mailer:
        enabled: true
        transports:
            main: sendmail://default
            pimcore_newsletter: sendmail://default
```
